src/test/java/com/example/service/WeatherServiceTest.java

```diff
@@ -0,0 +1,93 @@
+package com.example.service;
+
+import com.example.weatherwaveapi.model.request.WeatherRequest;
+import com.example.weatherwaveapi.model.response.WeatherApiResponse;
+import com.example.weatherwaveapi.model.response.WeatherForecastResponse;
+import com.example.weatherwaveapi.model.response.WeatherResponse;
+import com.example.weatherwaveapi.service.WeatherService;
+import com.example.weatherwaveapi.util.urlbuilder.OpenWeatherUrlBuilder;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.springframework.http.HttpMethod.GET;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Stream;
+
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.when;
+import static com.example.TestUtil.*;
+@ExtendWith(MockitoExtension.class)
+public class WeatherServiceTest {
```

Comment:
Sorry, I didn’t quite understand, are we testing the interface here?

Reply:
No no, after an update that was renamed into OpenWeatherServiceTest